spark/src/main/scala/io/delta/tables/DeltaTableBuilder.scala

```diff
@@ -280,6 +281,23 @@ class DeltaTableBuilder private[tables](
     this
   }
 
+  /**
+   * :: Evolving ::
+   *
+   * Specify the columns to cluster the output on the file system.
+   *
+   * Note: This should only include table columns already defined in schema.
+   *
+   * @param colNames string* column names for clustering
+   * @since 3.2.0
+   */
+  @Evolving
+  @scala.annotation.varargs
+  def clusterBy(colNames: String*): DeltaTableBuilder = {
+    clusteringColumns = Option(colNames)
```

Comment:
```suggestion
    clusteringColumns = Option(colNames)
    validatePartitioning()
```

We can move the check inside validatePartitioning and call it also in `partitionedBy` above. validatePartitioning can be:
```
  private def validatePartitioning(): Unit = {
    if (partitioningColumns.nonEmpty && clusteringColumns.nonEmpty) {
      throw DeltaErrors.clusterByWithPartitionedBy()
    }
  }
```

Reply:
Works for me